cpp/foxglove/src/channel.cpp

```diff
@@ -0,0 +1,29 @@
+#include <foxglove-c/foxglove-c.h>
+#include <foxglove/channel.hpp>
+
+namespace foxglove {
+
+Channel::Channel(std::string topic, std::string messageEncoding, std::optional<Schema> schema)
+    : _impl(
+        foxglove_channel_create(
+          topic.c_str(), messageEncoding.c_str(),
+          schema ? &((const foxglove_schema&)foxglove_schema{
+                     schema->name.data(),
+                     schema->encoding.data(),
+                     reinterpret_cast<const uint8_t*>(schema->data),
+                     schema->dataLen,
+                   })
+                 : nullptr
+        ),
+        foxglove_channel_free
+      ) {}
+
+void Channel::log(
+  const std::byte* data, size_t dataLen, uint64_t logTime, uint64_t publishTime, uint32_t sequence
```

Comment:
Again you'll want some way to omit these time and sequence arguments. In C++ you could use function overloading maybe.